salt/runners/salt.py

```diff
@@ -85,7 +86,13 @@ def cmd(fun, *args, **kwargs):
     with_pillar = kwargs.pop('with_pillar', False)
 
     opts = copy.deepcopy(__opts__)
-    opts['grains'] = salt.loader.grains(opts)
+    # try to only load grains if we need to, it may already exist from other contexts (ie pillar)
+    if 'grains' not in opts:
+        id_, grains, _ = salt.utils.minions.get_minion_data(__opts__['id'], __opts__)
```

Comment:
Is anything supposed to happen with `id_` here?

Reply:
nope. i can null that